python/rmm/_lib/device_buffer.pxd

```diff
@@ -42,9 +42,11 @@ cdef extern from "rmm/device_buffer.hpp" namespace "rmm" nogil:
 cdef class DeviceBuffer:
     cdef unique_ptr[device_buffer] c_obj
 
-    # Holds a reference to the device_memory_resource used for allocation.
-    # Ensures the memory resource does not get destroyed before
-    # this DeviceBuffer as it is needed for deallocation.
+    # Holds a reference to the memory resource used for allocation
+    # and ensures that it is not destroyed before the device buffer
+    # is deallocated. A reference to a DeviceMemoryResource cannot
+    # be used as it could be released prematurely because of the circle
+    # reference breaking mechanism.
```

Comment:
```suggestion
    # be used as it could be released prematurely by the gc if the DeviceBuffer
    # is in a reference cycle.
    # See https://github.com/rapidsai/rmm/pull/931 for details.
```